### ISSUE TITLE
Have you ever encountered this phenomenon: attr loss down to near 0 and rec loss keeps at 0.01 and doesn't got down?

### ISSUE THREAD

**Issue by buzhangjiuzhou** [mindslab-ai/faceshifter#20]
![image](https://user-images.githubusercontent.com/19666246/123499234-cfba4a00-d667-11eb-879d-440f17a5660a.png)
![image](https://user-images.githubusercontent.com/19666246/123499238-d5b02b00-d667-11eb-9ab3-4665dd36b810.png)

And the results :
![image](https://user-images.githubusercontent.com/19666246/123499245-e82a6480-d667-11eb-8af0-263aba17ffdb.png)

And before this, I trained with the origin setup of you, it run well before 400k steps, and at that point, the attr loss suddenly went down to near 0, at the same time, the rec loss went to 0.01 from 1e-3,  and the results ruined.

**Comment by kyugorithm** [comment]
Hi, I think your model has a problem with mode collapse.
I think we bumped into each other. G or D no longer competitive learning and fell into the wrong minima.

It is thought that a problem occurs when the batch_size is set small or the dataset is small while the capacity of the model is high.

How about increasing the data size or increasing the batch_size?
Or I think it would be good to find various solutions on Google.

**Comment by buzhangjiuzhou** [comment]
> Hi, I think your model has a problem with mode collapse. I think we bumped into each other. G or D no longer competitive learning and fell into the wrong minima.
> 
> It is thought that a problem occurs when the batch_size is set small or the dataset is small while the capacity of the model is high.
> 
> How about increasing the data size or increasing the batch_size? Or I think it would be good to find various solutions on Google.

I realized the problem and decreased the learning rate(because of the limitation of devices I can access). 

It helped.
